packages/blade/src/components/FileUpload/FileUpload.web.tsx

```diff
@@ -0,0 +1,323 @@
+    if (uploadType === 'single' && inputFiles.length > 1) {
+      setErrorMessage('You can upload only one file.');
+      setInternalValidationState('error');
+      return true;
+    }
+
+    if (maxCount && allFiles.length > maxCount) {
+      setErrorMessage(`You can't upload more than ${maxCount} files.`);
+      setInternalValidationState('error');
+      return true;
+    }
+
+    if (maxSize && inputFiles.some((file) => file.size > maxSize)) {
+      setErrorMessage('File size exceeded.');
+      setInternalValidationState('error');
+      return true;
+    }
+
+    setInternalValidationState('none');
+    setErrorMessage('');
+    return false;
+  };
+
+  const handleDragOver = (event: React.DragEvent): void => {
+    event.preventDefault();
+    setIsActive(true);
+  };
+
+  const handleDragLeave = (event: React.DragEvent): void => {
+    event.preventDefault();
+    setIsActive(false);
+  };
+
+  const handleDrop = (event: React.DragEvent): void => {
+    event.preventDefault();
+    setIsActive(false);
+
+    const droppedFiles = Array.from(event.dataTransfer.files);
+    const allFiles = selectedFiles.length > 0 ? [...selectedFiles, ...droppedFiles] : droppedFiles;
+
+    const hasValidationErrors = validateFiles(droppedFiles, allFiles);
+
+    if (!hasValidationErrors) {
+      handleFilesChange(droppedFiles);
+      onDrop?.({ name, fileList: allFiles });
+    }
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const inputFiles = Array.from(event.target.files ?? []);
+    const allFiles = selectedFiles.length > 0 ? [...selectedFiles, ...inputFiles] : inputFiles;
+
+    const hasValidationErrors = validateFiles(inputFiles, allFiles);
+
+    if (!hasValidationErrors) {
+      handleFilesChange(inputFiles);
+      onChange?.({ name, fileList: allFiles });
+    }
+  };
+
+  return (
+    <BaseBox
+      display="inline-flex"
+      {...metaAttribute({ name: MetaConstants.FileUpload, testID })}
+      {...getStyledProps(styledProps)}
+      width="100%"
+    >
+      <BaseBox
+        display="flex"
+        flexDirection={isLabelLeftPositioned ? 'row' : 'column'}
+        alignItems={isLabelLeftPositioned ? 'center' : undefined}
+        position="relative"
+        width="100%"
+      >
+        {label ? (
+          <FormLabel
+            as="span"
+            necessityIndicator={necessityIndicator}
+            position={labelPosition}
+            id={labelId}
+            accessibilityText={accessibilityText}
+          >
+            {label}
+          </FormLabel>
+        ) : null}
+
+        <BaseBox display="flex" flexDirection="column" marginBottom="spacing.5">
+          <SelectorLabel
+            componentName={MetaConstants.FileUploadLabel}
+            inputProps={{}}
+            style={{
+              cursor: isDisabled ? 'not-allowed' : 'pointer',
+              ...(!isMultiple && selectedFiles.length === 1 ? screenReaderStyles : {}),
```

Comment:
`!isMultiple && selectedFiles.length === 1`

Factor out and name this conditional with a variable.

Reply:
done